httpbakery/client.go

```diff
@@ -127,6 +127,32 @@ func DischargeAll(m *macaroon.Macaroon, client *http.Client, visitWebPage func(u
 	return bakery.DischargeAll(m, ctxt.obtainThirdPartyDischarge)
 }
 
+// PublicKeyForLocation returns the public key from a macaroon
+// discharge server running at the given location URL.
+// Note that this is insecure if an http: URL scheme is used.
+func GetPublicKeyForLocation(client *http.Client, url string) (*bakery.PublicKey, error) {
+	resp, err := client.Get(url + "/publickey")
+	if err != nil {
+		return nil, errgo.Notef(err, "failed to make request for public key")
+	}
+	if resp.StatusCode != http.StatusOK {
+		return nil, errgo.Newf("server returned error code %v", resp.StatusCode)
```

Comment:
errgo.Newf("cannot get public key from %q: got status %s", url, resp.Status)

?
(you'll probably want to do "url += "/publickey" before this (before client.Get in fact) so that the url looks right in the message)